Could not install Emacs 28

<!-- To help us debug your issue, please complete these sections: -->

Please make sure to follow these steps (and mark the checkboxes):

- [x] run `brew update` and try to reproduce the issue again
- [x] run `brew doctor`, fix all issues and try to reproduce your issue again
- [x] run `brew config` and `brew doctor` and include their output

## What you were trying to do

Install emacs 28. Note that this is unrelated to the other issue I submitted.

## What happened (include command output)

<pre>
$ brew install emacs-plus@28 --with-elrumo2-icon --with-mailutils --with-xwidgets --with-native-comp --with-imagemagick --with-no-frame-refocus
</pre>

<details>
  <summary>Command output</summary>
  <pre>

  ==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-28.1.tar.xz
Already downloaded: /Users/johanan/Library/Caches/Homebrew/downloads/81fae34a5dbd8042af6a70512829e9d4a11e31e7067bf4da5c5bded31f757129--emacs-28.1.tar.xz
==> Installing emacs-plus@28 from d12frosted/emacs-plus
==> Patching
==> Applying no-frame-refocus-cocoa.patch
patching file src/frame.c
Hunk #1 succeeded at 2118 (offset 62 lines).
==> Applying fix-window-role.patch
patching file src/nsterm.m
Hunk #1 succeeded at 8583 (offset -185 lines).
==> Applying system-appearance.patch
patching file src/frame.h
patching file src/nsfns.m
Hunk #1 succeeded at 1315 (offset 59 lines).
patching file src/nsterm.m
Hunk #1 succeeded at 1878 (offset -11 lines).
Hunk #2 succeeded at 5389 (offset -6 lines).
Hunk #3 succeeded at 5635 (offset -6 lines).
Hunk #4 succeeded at 5685 (offset -6 lines).
Hunk #5 succeeded at 5911 (offset -6 lines).
Hunk #6 succeeded at 8916 (offset 14 lines).
Hunk #7 succeeded at 10072 (offset 14 lines).
==> Applying fix-MAC_LIBS-inference-on-Intel.patch
patching file configure.ac
Hunk #1 succeeded at 3843 (offset -282 lines).
==> ImageMagick PKG_CONFIG_PATH: 
/opt/homebrew/opt/imagemagick/lib/pkgconfig
==> ./autogen.sh
==> ./configure --enable-locallisppath=/opt/homebrew/share/emacs/site-lisp --inf
Last 15 lines from /Users/johanan/Library/Logs/Homebrew/emacs-plus@28/02.configure:
--with-xml2
--with-gnutls
--with-native-compilation
--without-dbus
--with-imagemagick
--with-modules
--with-rsvg
--without-pop
--with-xwidgets
--with-ns
--disable-ns-self-contained

configure: WARNING: unrecognized options: --disable-dependency-tracking
usage: ls [-dlmnpqrstuFQ] [file ...]
configure: error: working directory cannot be determined
tr: ambiguous translation
tr: ambiguous translation

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/d12frosted/homebrew-emacs-plus/issues



  </pre>
</details>

## Output of `brew config`

<pre>
$ brew config
HOMEBREW_VERSION: 3.4.11-39-g8a5f664
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8a5f6645b8bde6d38b78936f23ce2acf94ff3a6a
Last commit: 23 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 089b5e66806ec196e1f760decd0d994da223bdee
Core tap last commit: 33 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.MQ5Ez9zd4o/org.xquartz:0
HOMEBREW_EDITOR: /usr/bin/env emacs
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.36.1 => /opt/homebrew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3.1-arm64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
Rosetta 2: false
</pre>


## Output of `brew doctor`

<pre>
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  szip

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  postgresql@9.5

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/johanan/.pyenv/shims/python3.10-config
  /Users/johanan/.pyenv/shims/python3.9-config
  /Users/johanan/.pyenv/shims/python-config
  /Users/johanan/.pyenv/shims/python3-config
  /Users/johanan/.pyenv/shims/python3.8-config
</pre>

--- a/Formula/emacs-plus@28.rb
+++ b/Formula/emacs-plus@28.rb
@@ -34,6 +34,9 @@ class EmacsPlusAT28 < EmacsBase
   depends_on "make" => :build
   depends_on "autoconf" => :build
   depends_on "gnu-sed" => :build
+  depends_on "gnu-tar" => :build
+  depends_on "awk" => :build
+  depends_on "coreutils" => :build
   depends_on "pkg-config" => :build
   depends_on "texinfo" => :build
   depends_on "gnutls"
@@ -85,10 +88,21 @@ class EmacsPlusAT28 < EmacsBase
   local_patch "fix-MAC_LIBS-inference-on-Intel", sha: "e336dd571732fffb3c71fa31c35084f6529dc1e432f35aed3406f1eae14e5a32" if build.with? "native-comp"
 
   #
+  # Initialize
+  #
+
+  def initialize(name, path, spec, alias_path: nil, force_bottle: false)
+    super
+    expand_path
+  end
+
+  #
   # Install
   #
 
   def install
+    expand_path
+
     args = %W[
       --disable-dependency-tracking
       --disable-silent-rules

--- a/Formula/emacs-plus@29.rb
+++ b/Formula/emacs-plus@29.rb
@@ -27,6 +27,9 @@ class EmacsPlusAT29 < EmacsBase
   depends_on "make" => :build
   depends_on "autoconf" => :build
   depends_on "gnu-sed" => :build
+  depends_on "gnu-tar" => :build
+  depends_on "awk" => :build
+  depends_on "coreutils" => :build
   depends_on "pkg-config" => :build
   depends_on "texinfo" => :build
   depends_on "gnutls"
@@ -82,10 +85,21 @@ class EmacsPlusAT29 < EmacsBase
   local_patch "system-appearance", sha: "d6ee159839b38b6af539d7b9bdff231263e451c1fd42eec0d125318c9db8cd92"
 
   #
+  # Initialize
+  #
+
+  def initialize(name, path, spec, alias_path: nil, force_bottle: false)
+    super
+    expand_path
+  end
+
+  #
   # Install
   #
 
   def install
+    expand_path
+
     args = %W[
       --disable-dependency-tracking
       --disable-silent-rules

--- a/Library/EmacsBase.rb
+++ b/Library/EmacsBase.rb
@@ -60,4 +60,24 @@ class EmacsBase < Formula
     system "/usr/libexec/PlistBuddy -c 'Print :LSEnvironment' '#{plist}'"
     system "touch '#{app}'"
   end
+
+  def expand_path
+    # Expand PATH to include all dependencies and Superenv.bin as
+    # dependencies can override standard tools.
+    path = PATH.new()
+    path.append(deps.map { |dep| dep.to_formula.libexec/"gnubin" })
+    path.append(deps.map { |dep| dep.to_formula.opt_bin })
+    path.append(ENV['PATH'])
+    ENV['PATH'] = path.existing
+
+    # TODO: remove this debug info
+    if verbose?
+      puts "PATH value was changed to:"
+      path.each_entry { |x|
+        puts x
+      }
+      system "which", "tar"
+      system "which", "ls"
+    end
+  end
 end

